node/libs/protobuf_build/src/lib.rs

```diff
@@ -52,70 +56,68 @@ fn traverse_files(
     Ok(())
 }
 
-/// Protobuf descriptor + info about the mapping to rust code.
+/// Manifest of a Protobuf compilation target containing information about the compilation process.
 #[derive(Debug)]
-pub struct Descriptor {
+struct Manifest {
     /// Root proto package that all proto files in this descriptor belong to.
-    /// Rust types have been generated relative to this root.
-    proto_root: ProtoName,
-    /// Raw descriptor proto.
-    descriptor_proto: prost_types::FileDescriptorSet,
-    /// Direct dependencies of this descriptor.
-    dependencies: Vec<&'static Descriptor>,
+    proto_root: ProtoPath,
+    /// Absolute path to the descriptor.
+    descriptor_path: String,
+    /// Tuples of `proto_root` and absolute paths to the corresponding descriptor for all dependencies
+    /// including transitive ones.
+    dependencies: Vec<(ProtoPath, String)>,
```

Comment:
std::path::Path?

Reply:
`PathBuf`, but yeah, this works better 👍 (This is an artifact from `serde` serialization of `Manifest`, btw.)